src/spikeinterface/core/generate.py

```diff
@@ -432,23 +444,68 @@ def __init__(
+                 duration: float,
+                sampling_frequency: float,
+                num_channels: int,
+                dtype: Optional[Union[np.dtype, str]] = "float32",
+                seed: Optional[int] = None,
+                mode: Literal['pure_noise', 'random_peaks'] = 'pure_noise'
+                ):
+        """
+        Initialize a GeneratorRecordingSegment instance.
+
+        This class is a subclass of BaseRecordingSegment and is used to generate synthetic recordings
+        with different modes, such as 'random_peaks' and 'pure_noise'.
+
+        Parameters
+        ----------
+        duration : float
+            The duration of the recording segment in seconds.
+        sampling_frequency : float
+            The sampling frequency of the recording in Hz.
+        num_channels : int
+            The number of channels in the recording.
+        dtype : numpy.dtype
+            The data type of the generated traces.
+        seed : int
+            The seed for the random number generator used in generating the traces.
+        mode : str
+            The mode of the generated recording, either 'random_peaks' or 'pure_noise'.
+        """
         BaseRecordingSegment.__init__(self, sampling_frequency=sampling_frequency)
         self.sampling_frequency = sampling_frequency
         self.num_samples = int(duration * sampling_frequency)
         self.seed = seed
         self.num_channels = num_channels
         self.dtype = dtype 
-
-        # Random numbers characterising the channels, need to be done outside for consistency
+        self.mode = mode
         self.rng = np.random.default_rng(seed=self.seed)        
-        self.channel_phases = self.rng.uniform(low=0, high=2*np.pi, size=self.num_channels)
-        self.frequencies = 1.0 + self.rng.exponential(scale=1.0, size=self.num_channels)
-        self.amplitudes = self.rng.normal(loc=70, scale=10.0, size=self.num_channels)
-        self.amplitudes *= self.rng.choice([-1, 1], size=self.num_channels)
+        
+        if self.mode == 'random_peaks':
+            self.traces_generator = self._random_peaks_generator
+            
+            # Configuration of mode
+            self.channel_phases = self.rng.uniform(low=0, high=2*np.pi, size=self.num_channels)
+            self.frequencies = 1.0 + self.rng.exponential(scale=1.0, size=self.num_channels)
+            self.amplitudes = self.rng.normal(loc=70, scale=10.0, size=self.num_channels)  # Amplitudes of 70 +- 10
+            self.amplitudes *= self.rng.choice([-1, 1], size=self.num_channels)    # Both negative and positive peaks
+        
+        elif self.mode == "pure_noise":
+            self.traces_generator = self._random_traces_generator
+
+            # Configuration of mode
+            noise_size_ms = 50.0
+            noise_size_frames = int(noise_size_ms * self.sampling_frequency / 1000)
+            self.basic_noise_block = self.rng.random(size=(noise_size_frames, self.num_channels))
```

Comment:
@h-mayorquin I discussed with @samuelgarcia and we agree it shoudl be `normal` instead. Note that the way it is it returns samples between 0 and 1!

Reply:
Yes, makes sense. I was trying to imitate the current `generate_recording`. Great.

Reply:
Done. Now the default mode is to generate `white_noise`.